sklearn/tree/_tree.pyx

```diff
@@ -247,6 +347,135 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
                 "lower_bound": -INFINITY,
                 "upper_bound": INFINITY,
             })
+            if rc == -1:
+                # got return code -1 - out-of-memory
+                raise MemoryError()
+
+        with nogil:
+            while not update_stack.empty():
```

Comment:
@PSSF23 does this code affect the normal building process not via `partial_fit`

Reply:
it shouldn't. update_stack only exists when the initial_roots has content.